tests/scale/virt-api.go

```diff
@@ -47,24 +96,31 @@ var _ = Describe("[sig-compute] virt-api scaling", decorators.SigCompute, func()
 
 	It("virt-api replicas should be scaled as expected", func() {
 		By("Finding out nodes count")
-		nodes, err := virtClient.CoreV1().Nodes().List(context.Background(), v1.ListOptions{})
-		Expect(err).ToNot(HaveOccurred())
-
-		numberOfNodes := len(nodes.Items)
-
+		expectedResult := calcExpectedReplicas(numberOfNodes)
 		Eventually(func() int32 {
-			By("Finding out virt-api replica number")
-			kv := util.GetCurrentKv(virtClient)
-			Expect(kv).ToNot(BeNil())
-
-			apiDeployment, err := virtClient.AppsV1().Deployments(kv.GetNamespace()).Get(context.Background(), components.VirtAPIName, v1.GetOptions{})
-			Expect(err).ToNot(HaveOccurred())
+			return getApiReplicas(virtClient, expectedResult)
+		}, 1*time.Minute, 5*time.Second).Should(Equal(calcExpectedReplicas(numberOfNodes)), "number of virt API should be as expected")
+	})
 
-			By("Expecting number of replicas to be as expected")
-			Expect(apiDeployment.Spec.Replicas).ToNot(BeNil())
+	It("[Serial]virt-api replicas should be determined by patch if exist", func() {
+		expectedResult := calcExpectedReplicas(numberOfNodes)
+		expectedResult += 1
```

Comment:
Is it necessary to calculate the numberOfNodes? Why can't you just use a predefined number? Thanks

Reply:
This way we ensure that the number set in the patch is distinct from the one calculated in calcExpectedReplicas. If these two numbers are identical, then this test would serve no purpose. 
Additionally, it's worth noting that calcExpectedReplicas does not involve any resource-intensive operations such as making API calls.